packages/twenty-front/src/modules/domain-manager/hooks/useRedirect.ts

```diff
@@ -0,0 +1,16 @@
+// Don't use this hook directly! Prefer the high level hooks like:
+// useRedirectToDefaultDomain and useRedirectToWorkspaceDomain
+import { sleep } from '~/utils/sleep';
+
+export const useRedirect = () => {
+  const redirect = (url: string) => {
+    // This hacky workaround is necessary to ensure the tokens stored in the cookie are updated correctly.
+    sleep(0).then(() => {
```

Comment:
useDebounced (100ms)

Reply:
const onMouseLeaveBoard = useDebouncedCallback(() => {
    if (isCompactModeActive && isCardExpanded) {
      setIsCardExpanded(false);
    }
  }, 800);

Reply:
name the hook differently as it's not a generic hook (it's tied to domain redirection): useDomainRedirect